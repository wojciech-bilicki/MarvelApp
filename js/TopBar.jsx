// @flow
import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import logo from '../assets/Marvel-Logo.png';

const TopBarWrapper = styled.header`
  width: 100%;
  background-color: #f11e22;
  display: flex;
  justify-content: space-between;
  align-items: center;
  span {
    color: white;
    font-size: 36px;
    font-weight: bold;
  }
`;

const Logo = styled.img`
  width: 200px;
  height: 80px;
  margin-left: 32px;
`;

const SearchInput = styled.div`
  margin-right: 32px;
  input {
    margin-right: 8px;
  }
`;

type State = {
  searchTerm: string
};

class TopBar extends React.Component<null, State> {
  state = {
    searchTerm: ''
  };

  onInputChange = (event: KeyboardEvent & { target: HTMLInputElement }) => {
    this.setState({ searchTerm: event.target.value });
  };

  render() {
    return (
      <TopBarWrapper>
        <Logo src={`${logo}`} alt="Logo" />
        <span>Super Marvel Teams</span>
        <SearchInput>
          <input value={this.state.searchTerm} onChange={this.onInputChange} />
          <Link to={`/search/${this.state.searchTerm}`}>
            <button>search hero</button>
          </Link>
        </SearchInput>
      </TopBarWrapper>
    );
  }
}

export default TopBar;
