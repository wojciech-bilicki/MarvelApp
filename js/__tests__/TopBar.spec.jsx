import React from 'react';
import { shallow } from 'enzyme';
import TopBar from '../TopBar';

jest.mock('react-router-dom', () => ({
  Link: 'Link'
}));

describe('TopBar', () => {
  let topBar = null;

  beforeEach(() => {
    topBar = shallow(<TopBar />);
  });

  it('should render properly', () => {
    expect(topBar).toMatchSnapshot();
  });

  it('should change the state on input', () => {
    topBar.find('input').simulate('change', { target: { value: 'Hulk' } });
    expect(topBar.state('searchTerm')).toEqual('Hulk');
  });
});
