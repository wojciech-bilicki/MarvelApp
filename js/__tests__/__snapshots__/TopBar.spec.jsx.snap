// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar should render properly 1`] = `
<styled.header>
  <styled.img
    alt="Logo"
    src="test-file-stub"
  />
  <span>
    Super Marvel Teams
  </span>
  <styled.div>
    <input
      onChange={[Function]}
      value=""
    />
    <Link
      to="/search/"
    >
      <button>
        search
      </button>
    </Link>
  </styled.div>
</styled.header>
`;
